crates/rpc/src/jsonrpc/router/subscription.rs

```diff
@@ -220,6 +228,22 @@ where
 
             db.block_number(starting_block)
                 .map_err(RpcError::InternalError)?
+                .map(|starting_block| -> Result<_, RpcError> {
+                    match max_history {
+                        WebsocketHistory::Limited(limit) => {
+                            let latest = db
+                                .block_number(pathfinder_storage::BlockId::Latest)
+                                .map_err(RpcError::InternalError)?
+                                .unwrap_or(BlockNumber::GENESIS);
+                            let starting_block_limited = latest.saturating_sub(limit);
+
+                            // Pick the one closer to the latest block.
+                            Ok(std::cmp::max(starting_block, starting_block_limited))
```

Comment:
So when configured with a limit of N we just take the input and map it into the valid range.

I think that's a bit counter-intuitive (when requesting a given range you cannot be sure if you're receiving all data) and also inconsistent with what Juno's doing (returning [TOO_MANY_BLOCKS_BACK](https://github.com/starkware-libs/starknet-specs/blob/a2d10fc6cbaddbe2d3cf6ace5174dd0a306f4885/api/starknet_ws_api.json#L389) error).